client/src/document/ingredients/browser-compatibility-table/feature-row.tsx

```diff
@@ -425,6 +425,29 @@ function CompatCell({
   );
 }
 
+function ShrinkText(props: { text: string }) {
+  const text: string = props.text || "";
+  // 75% of the string text is a reasonable
+  // cutoff point, but could be improved
+  // if we find that 75% of string isn't
+  // looking nice
+  const rate: number = 0.75;
+  let cutoff: number = Math.floor(text.length * rate);
+  // split the original string into two strings
+  // 1: 75% of original string
+  // 2: remaining 25%
+  let leftSide: string = text.slice(0, cutoff);
+  let rightSide: string = text.slice(cutoff);
+  return (
+    <>
+      <code title={text}>
+        <span className="left-side">{leftSide}</span>
+        <span className="right-side">{rightSide}</span>
+      </code>
+    </>
+  );
+}
```

Comment:
This TS code needs some work. I can give a more detailed review if and when @schalkneethling has had a chance to review the total effect of this functionality.

Reply:
Yeah, looking back at this. I did define a few types, but the it definitely needs some work.

Reply:
Outside of the TypeScript updates. What do you think of how the functionality works? From a reader standpoint? @peterbe

Reply:
> Outside of the TypeScript updates. What do you think of how the functionality works? From a reader standpoint? @peterbe

Haven't had a chance to even look that far :)
Will do when time allows. But for now I lean on Schalk to take a first overview-look.

Reply:
> Haven't had a chance to even look that far :) Will do when time allows. But for now I lean on Schalk to take a first overview-look

Ok no worries :) thanks for the heads up.